demisto_sdk/commands/validate/tests/IF_validators_test.py

```diff
@@ -0,0 +1,332 @@
+    """
+    content_items: List[IncidentField] = [
+        create_incident_field_object(["content"], [False])
+    ]
+
+    results = IsValidContentFieldValidator().is_valid(content_items)
+    assert results[0].message == "The `content` key must be set to true."
+
+
+def test_IsValidSystemFlagValidator_not_valid():
+    """
+    Given:
+        - IncidentField content items
+    When:
+        - run is_valid method
+    Then:
+        - Ensure that the ValidationResult returned
+          for the IncidentField whose 'system' field is set to True
+    """
+    content_items: List[IncidentField] = [
+        create_incident_field_object(["system"], [True])
+    ]
+
+    results = IsValidSystemFlagValidator().is_valid(content_items)
+    assert results[0].message == "The `system` key must be set to false."
+
+
+def test_IsValidFieldTypeValidator_not_valid():
+    """
+    Given:
+        - IncidentField content items
+    When:
+        - run is_valid method
+    Then:
+        - Ensure that the ValidationResult returned
+          for the IncidentField whose 'type' field is not valid
+    """
+    content_items: List[IncidentField] = [
+        create_incident_field_object(["type"], ["test"])
+    ]
+
+    results = IsValidFieldTypeValidator().is_valid(content_items)
+    assert (
+        results[0].message
+        == f"Type: `test` is not one of available types.\navailable types: {FIELD_TYPES}."
+    )
+
+
+def test_IsValidGroupFieldValidator_not_valid():
+    """
+    Given:
+        - IncidentField content items
+    When:
+        - run is_valid method
+    Then:
+        - Ensure that the ValidationResult returned
+          for the IncidentField whose 'group' field is not valid
+    """
+    content_items: List[IncidentField] = [create_incident_field_object(["group"], [2])]
+
+    results = IsValidGroupFieldValidator().is_valid(content_items)
+    assert results[0].message == "Group 2 is not a group field."
+
+
+def test_IsCliNameFieldAlphanumericValidator_not_valid():
+    """
+    Given:
+        - IncidentField content items
+    When:
+        - run is_valid method
+    Then:
+        - Ensure that the ValidationResult returned
+          for the IncidentField whose 'cliName' value is non-alphanumeric
+    """
+    content_items: List[IncidentField] = [
+        create_incident_field_object(["cliName"], ["test_1234"])
+    ]
+
+    results = IsCliNameFieldAlphanumericValidator().is_valid(content_items)
+    assert (
+        results[0].message
+        == "Field `cliName` contains non-alphanumeric or uppercase letters."
+    )
+
+
+def test_IsCliNameReservedWordValidator_not_valid():
+    """
+    Given:
+        - IncidentField content items
+    When:
+        - run is_valid method
+    Then:
+        - Ensure that the ValidationResult returned
+          for the IncidentField whose 'cliName' value is a reserved word
+    """
+    content_items: List[IncidentField] = [
+        create_incident_field_object(["cliName"], ["parent"])
```

Comment:
Please parameterize, using the constant of prevented values, rather than just hardcoding "parent"